src/librepgp/stream-armor.cpp

```diff
@@ -574,24 +574,48 @@ armor_parse_header(pgp_source_t *src)
     return true;
 }
 
+static bool
+armor_skip_line(pgp_source_t *src)
+{
+    char header[1024] = {0};
+    do {
+        size_t hdrlen = 0;
+        bool   res = src_peek_line(src, header, sizeof(header), &hdrlen);
+        if (hdrlen) {
+            src_skip(src, hdrlen);
+        }
+        if (res || (hdrlen < sizeof(header) - 1)) {
+            return res;
+        }
+    } while (1);
+}
+
 static bool
 armor_parse_headers(pgp_source_t *src)
 {
     pgp_source_armored_param_t *param = (pgp_source_armored_param_t *) src->param;
     char                        header[1024] = {0};
-    size_t                      hdrlen;
-    char *                      hdrval;
 
     do {
+        size_t hdrlen = 0;
         if (!src_peek_line(param->readsrc, header, sizeof(header), &hdrlen)) {
-            RNP_LOG("failed to peek line");
-            return false;
-        }
-        if (!hdrlen) {
+            /* if line is too long let's cut it to the reasonable size */
+            src_skip(param->readsrc, hdrlen);
+            if ((hdrlen < sizeof(header) - 1) || !armor_skip_line(param->readsrc)) {
```

Comment:
I'd suggest to replace the first condition with `(hdrlen != sizeof(header) - 1)`.
It will be evident that subsequent `header[hdrlen] = '\0'` is correct